### PR TITLE
Fix JS

### PR DESCRIPTION
**Pull request by wanlwanl**

### CODE REVIEW

File: src/SignalRServiceExtension/Bindings/SignalRInputBindings/SignalRValueProvider.cs [hunk 1]
```diff
@@ -10,10 +10,14 @@ namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
     internal class SignalRValueProvider : IValueProvider
     {
         private object value;
+        private string invokeString;
 
-        public SignalRValueProvider(object value)
+        // todo: fix invoke string in another PR
+        public SignalRValueProvider(object value, Type type, string invokeString)
```
Comment (JialinXin) [inline src/SignalRServiceExtension/Bindings/SignalRInputBindings/SignalRValueProvider.cs:16]: >string invokeString [](start = 61, length = 19)

Why add type/invokeString into parameter? couldn't understand its usage. JS is blocked in InputBindingProvider attribute resolver, anything related to here?

Comment (wanlwanl) [reply src/SignalRServiceExtension/Bindings/SignalRInputBindings/SignalRValueProvider.cs:16]: The js require the `IValueProvider` returned by `IBinding` has `Type` property that is a `string`. It should be the concrete type of the value of the `IValueProvider`. The attribute resolver is used for extract the final value of the binding expression, e.g. get a real user ID from http request: UserId="{req.header.x-client-userId}"

Comment (JialinXin) [reply src/SignalRServiceExtension/Bindings/SignalRInputBindings/SignalRValueProvider.cs:16]: I see.

---
In reply to: [411085354](https://github.com/Azure/azure-functions-signalrservice-extension/pull/119#discussion_r411085354) [](ancestors = 411085354)

File: src/SignalRServiceExtension/Bindings/SignalRInputBindings/SignalRConnectionInputBinding.cs [hunk 2]
```diff
@@ -64,7 +67,21 @@ internal class SignalRConnectionInputBinding : BindingBase<SignalRConnectionInfo
             signalRConnectionInfoConfigurer.Configure(tokenResult, request, signalRConnectionDetail);
             var customizedInfo = azureSignalRClient.GetClientConnectionInfo(signalRConnectionDetail.UserId,
                 signalRConnectionDetail.Claims);
-            return Task.FromResult((IValueProvider)new SignalRValueProvider(customizedInfo));
+            return Task.FromResult<IValueProvider>(new SignalRValueProvider(ConvertToUserType(customizedInfo), userType, ""));
```
Comment (JialinXin) [inline src/SignalRServiceExtension/Bindings/SignalRInputBindings/SignalRConnectionInputBinding.cs:70]: >ConvertToUserType(customizedInfo) [](start = 76, length = 33)

if real type (userType) is passed to SignalRValueProvider, shall this convert done inside SignalRValueProvider?

File: src/SignalRServiceExtension/Bindings/SignalRInputBindings/SignalRConnectionInputBinding/SignalRConnectionInfoValueProvider.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    internal class SignalRConnectionInfoValueProvider : IValueProvider
+    {
+        private object info;
```
Comment (JialinXin) [inline src/SignalRServiceExtension/Bindings/SignalRInputBindings/SignalRConnectionInputBinding/SignalRConnectionInfoValueProvider.cs:13]: >object [](start = 16, length = 6)

SignalRConnectionInfo?

**Pull request merged**